x-pack/plugin/core/src/main/java/org/elasticsearch/snapshots/SourceOnlySnapshotRepository.java

```diff
@@ -80,13 +80,14 @@
     public void finalizeSnapshot(SnapshotId snapshotId, ShardGenerations shardGenerations, long startTime, String failure,
                                  int totalShards, List<SnapshotShardFailure> shardFailures, long repositoryStateId,
                                  boolean includeGlobalState, MetaData metaData, Map<String, Object> userMetadata,
-                                 boolean writeShardGens, ActionListener<SnapshotInfo> listener) {
+                                 boolean writeShardGens, boolean writeIndexGens, ActionListener<SnapshotInfo> listener) {
```

Comment:
I kind of hate adding more flags here, maybe we could refactor this to having a `Version version` input after all in the next step? With our new BwC strategy, these flags are kind of here to stay I think. I don't really see us making a hard break in repo BwC in the near future.